frontend/src/component/releases/ReleasePlanTemplate/MilestoneList.tsx

```diff
@@ -0,0 +1,39 @@
+import type { IReleasePlanMilestonePayload } from 'interfaces/releasePlans';
+import { useState } from 'react';
+import { MilestoneCard } from './MilestoneCard';
+
+interface IMilestoneList {
+    milestones: IReleasePlanMilestonePayload[];
+    setAddStrategyOpen: (open: boolean) => void;
+    errors: { [key: string]: string };
+    clearErrors: () => void;
+}
+
+export const MilestoneList = ({
+    milestones,
+    setAddStrategyOpen,
+    errors,
+    clearErrors,
+}: IMilestoneList) => {
+    const [currentMilestone, setCurrentMilestone] = useState<number>(-1);
+
+    const showAddStrategyDialog = (index: number) => {
+        setCurrentMilestone(index);
+        setAddStrategyOpen(true);
+    };
+
+    return (
+        <>
+            {milestones.map((milestone, index) => (
```

Comment:
Try not to rely on the index here. It's going to be especially troublesome if we decide to allow the user to sort / change the order of the milestones.

Ideally you would have unique ids you could use for each milestone, however those are generated only in the backend. My suggestion is [creating temporary uuids](https://github.com/Unleash/unleash/pull/8768/files#diff-67383b99ce55a9de4c95252431abfc962e7410dc62fb3c43d5375ca289406d09R126) in the frontend for state management and omitting them from the final payload.

Reply:
I still think relying on indexes here will bite you down the line, but if you're willing to accept that then feel free to proceed for now.